compiler/src/iree/compiler/Dialect/Util/Analysis/Constant/ConstExpr.cpp

```diff
@@ -309,6 +309,32 @@ void ConstExprHoistingPolicy::initialize() {
   }
 }
 
+static bool doesHoistingIncreaseSizeSignificantly(
+    const ConstExprAnalysis::ConstValueInfo *info) {
+
+  int64_t inSizeBits = 0;
+  for (Value root : info->roots) {
+    // TODO: Are there any other types we care about here?
+    if (auto type = dyn_cast<ShapedType>(root.getType())) {
+      inSizeBits += type.getNumElements() * type.getElementTypeBitWidth();
+    }
+  }
+
+  int64_t outSizeBits = 0;
+  if (auto type = dyn_cast<ShapedType>(info->constValue.getType())) {
+    outSizeBits = type.getNumElements() * type.getElementTypeBitWidth();
+  }
+
+  // Anything smaller than 1MB is okay to hoist.
+  const int64_t kThresholdVerySmall = 1024 * 1024 * 8;
```

Comment:
If possible, this would be really nice to have control over with a flag IMHO (at least for being able to set it to 0 in certain cases).

Reply:
yeah, make a flag like we have https://github.com/openxla/iree/blob/main/compiler/src/iree/compiler/Dialect/Flow/Transforms/RegionOpUtils.cpp#L34-L38
(e.g. "iree-flow-const-expr-max-size-increase-threshold")